### ISSUE TITLE
Normalization during power iteration can lead to under-estimating Lipschitz constants (i.e. unsoundness) for small weights

### ISSUE THREAD

**Issue by tobycmurray** [klasleino/gloro#8]
During power iteration, after each iteration the vector is normalized by `gloro.utils.l2_normalize`, which is implemented as:

```python
def l2_normalize(x, axis=None):
    return x / (
        tf.sqrt(tf.reduce_sum(x**2., axis=axis, keepdims=True)) +
            gloro.constants.EPS)
```

When the vector being normalized is small, the addition of `gloro.constants.EPS` to the length (to guard against division-by-zero) can significantly inflate the length, thereby artificially reducing the size of the iterate and resulting in under-estimating the true Lipschitz constants.

For example, consider a single-neuron layer whose single weight is initialized to `1e-5`. In this case, power iteration on (assumed normalized) input vector `x` computes `1e-10 * x`, whose length is then `1e-10` and significantly smaller than `EPS=1e-9`. Thus the addition of `EPS` during normalization makes the divisor much larger than it should be. Then, power iteration under-estimates the true Lipschitz constant for this layer (which is of course `1e-5`). Instead, the current implementation (a218dcdaaa41951411b0d520581e96e7401967d7) computes `9.434563e-06`.

One way to fix this would be to add `EPS` during normalization only when the vector's length is 0, i..e:

```python
def l2_normalize(x, axis=None):
    rawnorm = tf.sqrt(tf.reduce_sum(x**2., axis=axis, keepdims=True))
    norm = tf.where(rawnorm > 0, rawnorm, rawnorm + gloro.constants.EPS)
    return x / norm
```

**Comment by tobycmurray** [comment]
Updated issue description to refer to the correct current HEAD revision a218dcdaaa41951411b0d520581e96e7401967d7 to which this issue pertains.
